### PR TITLE
[ADD] website: reCAPTCHA

### PR DESCRIPTION
**Pull request by xpl-odoo**
task-3336129

### CODE REVIEW

**Review by toaa-odoo — APPROVED** [review]

**Comment by xpl-odoo** [comment]
@Abridbus I think you can proceed with the review :)

File: content/applications/websites/website/configuration/recaptcha.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     subdomain.odoo.com).
+   - A valid domain requires a host and must not include any path, port, query, or fragment.
+
+- Under :guilabel:`Google Cloud Platform`, a project is automatically selected if one was already
+  created with the logged-in Google account. If not, one is automatically created. Click
+  :guilabel:`Google Cloud Platform` to select a project yourself or rename the automatically created
+  project.
+- Agree to the terms of service.
+- Click :guilabel:`Submit`.
+
```
[the remark this answers is not in the document]
Comment (xpl-odoo) [reply content/applications/websites/website/configuration/recaptcha.rst:40]: - Odoo handles the configuration; you don't use https://console.cloud.google.com/ or the reCAPTCHA Enterprise API
- reCAPTCHA Enterprise isn't the same as reCAPTCHA v3
- reCAPTCHA v3 is free, with no need for a paid "Google Cloud" account (up to 1 million assessment/month)

File: content/applications/websites/website/configuration/recaptcha.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+   `Google's reCAPTCHA v3 guide <https://developers.google.com/recaptcha/docs/v3>`_
+
+Configuration
+=============
+
+On Google
+---------
+
+Open `the reCAPTCHA website registration page <https://www.google.com/recaptcha/admin/create>`_. Log
+in or create a Google account if necessary.
```
[the remark this answers is not in the document]
Comment (xpl-odoo) [reply content/applications/websites/website/configuration/recaptcha.rst:21]: I can't find information on the "classic key".

File: content/applications/websites/website/configuration/recaptcha.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+On the website registration page:
+
+- Give the website any :guilabel:`Label`.
+- Leave the :guilabel:`reCAPTCHA type` on :guilabel:`Score based (v3)`.
+- Enter one or more :guilabel:`Domains`.
+
+.. note::
+   - Registering a domain name (e.g., odoo.com) also registers any subdomain (e.g.,
+     subdomain.odoo.com).
```
[the remark this answers is not in the document]
Comment (xpl-odoo) [reply content/applications/websites/website/configuration/recaptcha.rst]: I don't think it has an impact to use odoo.com. It seems limits are per project _limits apply to each Google Cloud console project and are shared across all applications and IP addresses using that project._

File: content/applications/websites/website/configuration/recaptcha.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+On Odoo
+-------
+
+- From the database dashboard, click :guilabel:`Settings`. Under :guilabel:`Integrations`, enable
+  :guilabel:`reCAPTCHA` if needed.
+
+  .. warning::
+     Do not disable the :guilabel:`reCAPTCHA` feature or uninstall the :guilabel:`Google reCAPTCHA
+     integration` module, as many other modules would be removed as well.
```
[the remark this answers is not in the document]
Comment (xpl-odoo) [reply content/applications/websites/website/configuration/recaptcha.rst:58]: Added that later (it works for Newsletter Block and Newsletter Popup as well). I might add a tip about the eCommerce extra step form

**Comment by xpl-odoo** [comment]
On 16.0: add a tip about eCommerce extra step forms
On 16.2: add content about Cloudflare Turnstile

**Comment by xpl-odoo** [comment]
@robodoo r+

**Comment by xpl-odoo** [comment]
@robodoo retry

**Pull request closed without merging**